client.go

```diff
@@ -477,13 +487,26 @@ func (c *Client) stream(method, path string, streamOptions streamOptions) error
 	if streamOptions.stderr == nil {
 		streamOptions.stderr = ioutil.Discard
 	}
-	cancelRequest := cancelable(c.HTTPClient, req)
+
+	// make a sub-context so that our active cancellation does not affect parent
+	ctx := streamOptions.context
+	if ctx == nil {
+		ctx = context.Background()
+	}
+	subCtx, cancelRequest := context.WithCancel(ctx)
+	defer cancelRequest()
+
 	if protocol == "unix" {
 		dial, err := c.Dialer.Dial(protocol, address)
 		if err != nil {
 			return err
 		}
-		cancelRequest = func() { dial.Close() }
+		go func() {
+			select {
+			case <-subCtx.Done():
+				dial.Close()
```

Comment:
Is there a problem if we Close the connection twice? I guess that could have happened in the old code too but it could have been buggy?